### ISSUE TITLE
Research R, Python, Anaconda Update/Optimization

### ISSUE THREAD

**Issue by bryanpaget** [StatCan/aaw-kubeflow-containers#496]
The following issues https://github.com/StatCan/aaw/issues/1784, https://github.com/StatCan/aaw-kubeflow-containers/issues/485 and https://github.com/StatCan/aaw-kubeflow-containers/pull/492 revealed a fundamental problem with the way Anaconda is installed in our images. We need to rethink the Anaconda installation.

Possible course of action:

- [ ] find any and all settings for Anaconda and set them to the version we want.
- [ ] find and move any and all Anaconda related installation and configuration steps to an Anaconda docker-bit shared by all images (Anaconda is currently being installed in both ∞_CMD.Dockerfile and ∞_CMD_remote-desktop.Dockerfile).
- [ ] I don't think we need to install rstudio-server. At least on the remote-desktop image we already have rstudio-server installed from the base image. See: https://github.com/StatCan/aaw-kubeflow-containers/issues/496#issuecomment-1689126741
- [ ] Some images should be upgraded to "2023-08-07".

![Image](https://github.com/StatCan/aaw-kubeflow-containers/assets/8212170/8f16a8f5-2346-43e1-ace7-3ccf3427f192)

**Comment by bryanpaget** [comment]
- Create a new docker-bit for Anaconda since it's currently being installed in both ∞_CMD.Dockerfile and ∞_CMD_remote-desktop.Dockerfile.

**Comment by bryanpaget** [comment]
Notes for Future Work:

- [ ] We have the tag for `jupyter/datascience-notebook` specific three times in the docker-bits. Next sprint let's refactor some more to have it listed only once. The files are: `0_cpu`, `0_cpu_sas` and the `Makefile`.

**Comment by bryanpaget** [comment]
This has proven to be quite difficult. Earlier in the sprint I tried refactoring the docker-bits files, with the goal of separating and isolating the Anaconda installation to its own Anaconda.dockerfile. This has not worked out.

Not working:

- https://github.com/StatCan/aaw-kubeflow-containers/actions/runs/5868964608

Latest attempt:

- https://github.com/StatCan/aaw-kubeflow-containers/actions/runs/5882103018

**Comment by bryanpaget** [comment]
Blocked by https://github.com/StatCan/aaw-kubeflow-containers/issues/502

**Comment by bryanpaget** [comment]
I don't think we need to install rstudio-server:

I made the following update recently

![image](https://github.com/StatCan/aaw-kubeflow-containers/assets/8212170/ffa08a96-ef98-4a9e-8adc-4f6e2f817d9e)

but discovered the new version was not being used

![image](https://github.com/StatCan/aaw-kubeflow-containers/assets/8212170/0cf0ac4e-4250-4170-8c08-6b8f93fe819a)

as you can see the rstudio-server version is stuck at 2023.06.0-421. That version comes from 

![image](https://github.com/StatCan/aaw-kubeflow-containers/assets/8212170/6520e4a8-cdf2-4a34-89fa-c179b9c0b50f)

which pins rstudio-server at 2023.06.0-421

![image](https://github.com/StatCan/aaw-kubeflow-containers/assets/8212170/bb33ecb2-6fab-470c-89c7-f23f73ac698b)

**Comment by bryanpaget** [comment]
We had a discussion during the August 31 2023 tech elab, which resulted in the following EPIC: [#1822](https://github.com/StatCan/aaw/issues/1822).
